traversal/procedure/ProcedureEdge.java

```diff
@@ -939,19 +943,22 @@ public Forwardable<? extends ThingVertex, Order.Asc> branch(
                             GraphManager graphMgr, Vertex<?, ?> fromVertex, Traversal.Parameters params
                     ) {
                         assert fromVertex.isThing() && fromVertex.asThing().isAttribute();
-                        Forwardable<ThingVertex, Order.Asc> iter;
                         AttributeVertex<?> att = fromVertex.asThing().asAttribute();
 
                         if (to.props().hasIID()) {
-                            iter = backwardBranchToIIDFiltered(graphMgr, att, HAS, params.getIID(to.id().asVariable()), to.props().types());
+                            Optional<ThingVertex> toVertex = backwardBranchToIIDFiltered(graphMgr, att, HAS, params.getIID(to.id().asVariable()), to.props().types());
+                            if (toVertex.isPresent()) return to.iterateAndFilterPredicates(toVertex.get(), params, ASC);
+                            else return emptySorted();
                         } else {
                             Set<TypeVertex> owners = graphMgr.schema().ownersOfAttributeType(att.type());
-                            iter = iterate(owners).filter(owner -> to.props().types().contains(owner.properLabel()))
-                                    .mergeMapForwardable(t -> att.ins().edge(HAS, PrefixIID.of(t.encoding().instance()), t.iid()).from(), ASC);
+                            return to.mergeAndFilterPredicatesOnVertices(
```

Comment:
the same pattern applies throughout: the `to` Vertex will filter and iterate/merge vertices  generated by the edge